app/src/main/java/dev/arkbuilders/navigator/data/stats/category/TagLabeledNStorage.kt

```diff
@@ -27,8 +27,14 @@ class TagLabeledNStorage(
     override suspend fun init() {
         val storage = locateStorage()
         if (storage?.exists() == true) {
-            val json = Json.decodeFromStream<JsonTagLabeledN>(storage.inputStream())
-            tagLabeledAmount.putAll(json.data)
+            try {
+                val json = Json.decodeFromStream<JsonTagLabeledN>(
+                    storage.inputStream()
+                )
+                tagLabeledAmount.putAll(json.data)
+            } catch (exception: Exception) {
```

Comment:
It seems that just logging the exception can be harmful for the storage. We can have 2 kinds of exception here:
1. Exception during decoding the storage, i.e. reading from disk.
2. An unlikely problem during `putAll`.

In either case, the `tagLabeledAmount` mapping is incorrect. When new stats appear, the will be written to the disk, but only part of the original storage will be written. E.g. if we have 5 entries in the storage, we read 2 of them and then we have an exception. We continue execution with 2/5 entries. Then new entry is coming, so we write 3/6 to the disk. This means we lose some entries in case of the exception.

Stats storage is not that important comparing to tags or scores, but it enhances UX by sorting tags in last-used order. We should avoid losing this data if possible.